module/os/linux/zfs/zvol_os.c

```diff
@@ -513,7 +513,7 @@ zvol_request_impl(zvol_state_t *zv, struct bio *bio, struct request *rq,
 		blk_mq_hw_queue = rq->mq_hctx->queue_num;
 #else
 		blk_mq_hw_queue =
-		    rq->q->queue_hw_ctx[rq->q->mq_map[rq->cpu]]->queue_num;
+		    rq->q->queue_hw_ctx[rq->q->mq_map[get_cpu()]]->queue_num;
```

Comment:
```suggestion
		    rq->q->queue_hw_ctx[rq->q->mq_map[smp_processor_id()]]->queue_num;
```

`get_cpu() ` sneakily disables preemption until the matching `put_cpu()`.  Let's use `smp_processor_id()` here which just gets the cpu id.

Reply:
Sigh, of course it does. Thanks. Updated, tested, pushed.